torch/csrc/distributed/c10d/reducer.cpp

```diff
@@ -225,8 +226,18 @@ void Reducer::mark_variable_ready_dense(VariableIndex index) {
     // `detach_` from `zero_grad`, which is incompatible with views.
     TORCH_INTERNAL_ASSERT(!grad.is_alias_of(bucket_view));
     TORCH_INTERNAL_ASSERT(grad.device() == bucket_view.device());
-    TORCH_INTERNAL_ASSERT(grad.numel() == bucket_view.numel());
-    bucket_view.copy_(grad.view({-1}), /* non_blocking */ true);
+    TORCH_INTERNAL_ASSERT(grad.sizes() == bucket_view.sizes());
+    // AccumulateGrad doesn't HAVE to obey the grad layout contract.
+    // The penalty for disobedience is reduced performance, not numerical death.
+    // Warnings here help diagnose poor DDP performance.
+    if (grad.strides() != bucket_view.strides()) {
+      TORCH_WARN_ONCE("Grad strides do not match bucket view strides. ",
```

Comment:
Similar comment as the one in AccumulateGrad, adding the sizes/strides here might be helpful.